### PR TITLE
ussologin: pass thorugh Ubuntu SSO Errors

### PR DESCRIPTION
**Pull request by mhilton**
Pass through any errors that are returned by Ubuntu SSO when retrieving
a token.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/idmclient/50/
Test PASSed.

File: ussologin/ussologin.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ var (
 // information to obtain an OAuth token from Ubuntu SSO. The returned
 // token can subsequently be used with LoginWithToken to perform a login.
 // The tokenName argument is used as the name of the generated token in
-// Ubuntu SSO.
+// Ubuntu SSO. If Ubuntu SSO returned an error when trying the retrieve
```
Comment (mattyw) [inline ussologin/ussologin.go]: to retrieve

File: ussologin/ussologin.go [hunk 2]
```diff
@@ -37,7 +37,8 @@ var (
...
+// the token the error will have a cuase of type *usso.Error.
```
Comment (mattyw) [inline ussologin/ussologin.go]: cause

**Comment by mattyw** [comment]
:+1:

File: ussologin/ussologin.go (hunk 1 above)
Comment (mhilton) [inline ussologin/ussologin.go]: done

File: ussologin/ussologin.go (hunk 2 above)
Comment (mhilton) [inline ussologin/ussologin.go]: done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/idmclient/51/
Test PASSed.

**Comment by fabricematrat** [comment]
:+1:

**Comment by rogpeppe** [comment]
LGTM but maybe a test?

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/idmclient-merge

**Pull request merged**